### ISSUE TITLE
always clip loss is zero

### ISSUE THREAD

**Issue by ryhhtn** [rinnakk/japanese-clip#3]
I tried to calculate clip loss using this code.

```
from PIL import Image
import torch
import japanese_clip as ja_clip

device = "cuda" if torch.cuda.is_available() else "cpu"
# ja_clip.available_models()
# ['rinna/japanese-clip-vit-b-16', 'rinna/japanese-cloob-vit-b-16']
model, preprocess = ja_clip.load("rinna/japanese-clip-vit-b-16", cache_dir="/tmp/japanese_clip", device=device)
tokenizer = ja_clip.load_tokenizer()

image = preprocess(Image.open("./img/dog.jpeg")).unsqueeze(0).to(device)
encodings = ja_clip.tokenize(
    texts="象",
    max_seq_len=77,
    device=device,
    tokenizer=tokenizer, # this is optional. if you don't pass, load tokenizer each time
)

with torch.no_grad():
    res = model(input_ids=encodings['input_ids'], pixel_values=image, return_loss=True)
    
print("clip loss:", res.loss.item())
```

but, no matter how many times I change the image, always clip loss return 0.

If the usage seems to be wrong, would you tell me how to use it correctly?

**Comment by mkshing** [comment]
@ryhhtn Hi! The reason why is because the number of pairs is only 1 and the InfoNCE loss is always -log(1)=0. Please double-check how CLIP works. 
So, here's an example to compute the loss on 8 image-text pairs. Remember that this is just a minimal example since the mini-batch size is too small. 

```python
import os
import skimage
import IPython.display
import matplotlib.pyplot as plt
from PIL import Image
import numpy as np

import torch
import japanese_clip as ja_clip

# load models
device = "cuda" if torch.cuda.is_available() else "cpu"
# ja_clip.available_models()
# ['rinna/japanese-clip-vit-b-16', 'rinna/japanese-cloob-vit-b-16']
model, preprocess = ja_clip.load("rinna/japanese-clip-vit-b-16", cache_dir="/tmp/japanese_clip", device=device)
tokenizer = ja_clip.load_tokenizer()

# prepare 8 pairs 
descriptions = {
    "page": "セグメンテーションに関して書かれたページ",
    "chelsea": "トラ猫の顔写真",
    "astronaut": "アメリカ国旗の前に立つ宇宙飛行士",
    "rocket": "発射台に立つロケット",
    "motorcycle_right": "ガレージにある赤いバイク",
    "camera": "三脚の上のカメラを見る人",
    "horse": "馬の白黒のシルエット", 
    "coffee": "お皿に載ったコーヒーカップ"
}

original_images = []
images = []
texts = []
plt.figure(figsize=(16, 5))

for filename in [filename for filename in os.listdir(skimage.data_dir) if filename.endswith(".png") or filename.endswith(".jpg")]:
    name = os.path.splitext(filename)[0]
    if name not in descriptions:
        continue

    image = Image.open(os.path.join(skimage.data_dir, filename)).convert("RGB")
  
    plt.subplot(2, 4, len(images) + 1)
    plt.imshow(image)
    plt.title(f"{filename}\n{descriptions[name]}")
    plt.xticks([])
    plt.yticks([])

    original_images.append(image)
    images.append(preprocess(image))
    texts.append(descriptions[name])

plt.tight_layout()

# compute loss
image_input = torch.tensor(np.stack(images)).to(device)
encodings = ja_clip.tokenize(texts, max_seq_len=77, device=device, tokenizer=tokenizer)
encodings["pixel_values"] = image_input
encodings["return_loss"] = True
outputs = model(**encodings)
loss = outputs.loss
print(f"loss: {loss.item()}")
# 0.00013485984527505934
```

**Comment by mkshing** [comment]
@ryhhtn I will close this issue. Please reopen it if you have more questions.
